marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -249,10 +249,15 @@ private String initializeJob(StructType schema) {
             if(options.get("additionalexternalmetadata")!=null) {
                 externalMetadata = (ObjectNode) objectMapper.readTree(options.get("additionalexternalmetadata"));
             }
+        } catch(Exception e) {
+                throw new IllegalArgumentException("Unable to parse additionalExternalMetadata option as a JSON object; " +
+                    "cause: " + e.getMessage(), e);
+        }
+        try{
             externalMetadata.set("sparkSchema", objectMapper.readTree(schema.json()));
 
         } catch (Exception e) {
-            logger.warn("Unable to read Spark schema as a JSON object; cause: " + e.getMessage() + "; the schema will not " +
+            throw new IllegalArgumentException("Unable to read Spark schema as a JSON object; cause: " + e.getMessage() + "; the schema will not " +
```

Comment:
I think this should still be a warning and not cause an error. The reasoning is that Ernie can fix a problem with additionalexternalmetadata - he's in control of that. But if for some reason we can't read the Spark StructType as JSON, it's not likely that Ernie can fix that, and now he can't ingest any data.

Reply:
Done